framework/db/QueryBuilder.php

```diff
@@ -1233,9 +1233,10 @@ public function getColumnType($type)
      */
     public function buildSelect($columns, &$params, $distinct = false, $selectOption = null)
     {
-        $select = $distinct ? 'SELECT DISTINCT' : 'SELECT';
-        if ($selectOption !== null) {
-            $select .= ' ' . $selectOption;
+        $select = $selectOption !== null ? 'SELECT ' . $selectOption : 'SELECT';
```

Comment:
what's DB/version you use? cant find sql syntax where something stay before 'distinct' in 'select' statement